lib/IDE/SyntaxModel.cpp

```diff
@@ -890,15 +906,18 @@ bool ModelASTWalker::walkToDeclPre(Decl *D) {
   } else if (auto *ED = dyn_cast<ExtensionDecl>(D)) {
     // Normally bindExtension() would take care of computing the extended
     // nominal. It must be done before asking for generic parameters.
-    ED->computeExtendedNominal();
+    if (!inInactiveClause)
+      ED->computeExtendedNominal();
     SyntaxStructureNode SN;
     setDecl(SN, D);
     SN.Kind = SyntaxStructureKind::Extension;
     SN.Range = charSourceRangeFromSourceRange(SM, ED->getSourceRange());
     SN.BodyRange = innerCharSourceRangeFromSourceRange(SM, ED->getBraces());
     SourceRange NSR = SourceRange();
-    if (auto *repr = ED->getExtendedTypeRepr())
-      NSR = repr->getSourceRange();
+    if (!inInactiveClause || ASTScope::areInactiveIfConfigClausesSupported()) {
+      if (auto *repr = ED->getExtendedTypeRepr())
```

Comment:
Retrieving the type representation should be purely syntactic; I don't think this change should be needed.

Reply:
The compiler crashes without it. Help me out here: what would be a better fix?

Reply:
@DougGregor Thanks to @nathawes , I realized that I completely misunderstood your comment here. I've removed this change. Thanks for your help and patience.